hyper/common/exceptions.py

```diff
@@ -71,3 +71,10 @@ class MissingCertFile(Exception):
     The certificate file could not be found.
     """
     pass
+
+
+class ProxyError(Exception):
```

Comment:
There is a `hyper.common.exceptions.SocketError`, which is not referenced anywhere in the project. It's very similar to the `ConnectionError` I've just introduced, I guess.

I suggest to leave them both here for now, but I would like to work on exceptions structure in another PR after this one